test/e2e/operator_removal_test.go

```diff
@@ -0,0 +1,238 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorapi "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned"
+	clusteroperatorhelpers "github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/openshift/vmware-vsphere-csi-driver-operator/test/library"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	clientName        = "vsphere-operator-e2e"
+	storageCOName     = "storage"
+	csiDriverName     = "csi.vsphere.vmware.com"
+	disabledCondition = "VMwareVSphereControllerDisabled"
+
+	csiDriverNameSpace          = "openshift-cluster-csi-drivers"
+	csiControllerDeploymentName = "vmware-vsphere-csi-driver-controller"
+	csiNodeDaemonSetName        = "vmware-vsphere-csi-driver-node"
+)
+
+type disableConditionStatus int
+
+const (
+	operatorDisabled = iota
+	operatorEnabled
+	clusterCSIDriverNotFound
+)
+
+var (
+	WaitPollInterval = time.Second
+	WaitPollTimeout  = 10 * time.Minute
+)
+
+func TestStorageRemoval(t *testing.T) {
```

Comment:
Can we add explicit comment before `func TestStorageRemoval` that this test does not check if any of static resources or conditions removed? I understand it would be tedious to check all resources and conditions, but may be someone would desire to increase coverage adding them in future.

Reply:
added